stored_requests/events/database/database.go

```diff
@@ -113,13 +114,22 @@ func (e *PostgresEventProducer) fetchAll() (fetchErr error) {
 	return nil
 }
 
-func (e *PostgresEventProducer) fetchDelta() (fetchErr error) {
+func (e *DatabaseEventProducer) fetchDelta() (fetchErr error) {
 	timeout := e.cfg.CacheUpdateTimeout * time.Millisecond
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
 
 	startTime := e.time.Now().UTC()
-	rows, err := e.cfg.DB.QueryContext(ctx, e.cfg.CacheUpdateQuery, e.lastUpdate)
+
+	var queryArgs []interface{}
+	if e.cfg.DBDriver == "mysql" {
+		queryArgs = append(queryArgs, e.lastUpdate, e.lastUpdate)
```

Comment:
`queryArgs ` expects 1 parameter, not 2. If I modify it to `queryArgs = append(queryArgs, e.lastUpdate)` all works good. There is no reason to pass two `e.lastUpdate`?

Reply:
The reason of passing the same parameter twice (in case of MySql) is in the difference of how it processes the parameters. Here are `poll_for_updates` queries for both cases:
mysql:
```
SELECT id, requestData, 'request' AS type FROM stored_requests WHERE last_updated > ?
UNION ALL
SELECT id, impData, 'imp' AS type FROM stored_imps WHERE last_updated > ?
```
postgres:
```
SELECT id, requestData, 'request' AS type FROM stored_requests WHERE last_updated > $1
UNION ALL
SELECT id, impData, 'imp' AS type FROM stored_imps WHERE last_updated > $1
```
Postgres references parameter by its number in the array of passed parameters, so one is enough.
MySql takes parameter from the passed array based on the number of question marks in the query. In other words, the number of parameters passed must equal to the number of question marks in the query.

Reply:
I see the problem now. 
Also in our set up we store "requests" and "imps" in the same table.  And yes like you said in postgres we can access parameter by number. 
May I discuss this with a team to get ideas how this can be solved and I'll get back to you.

Reply:
@VeronikaSolovei9 maybe any progress? we need mysql support a lot in here.